packages/matchbox/src/components/Pagination/Pagination.js

```diff
@@ -48,21 +48,23 @@ function Pagination(props) {
   }
 
   function _createButtons(start, end) {
-    const buttons = [];
-
-    for (let currentIndex = start; currentIndex < end; currentIndex++) {
-      buttons.push({
-        content: `${currentIndex + 1}`,
-        onClick: handlePageChange.bind(this, currentIndex),
-        flat: currentIndex === index ? false : true,
-        size: 'small',
-        color: currentIndex === index ? 'blue' : 'gray',
-        mx: '100',
-        width: currentIndex + 1 < 100 ? '32px' : 'auto',
-      });
-    }
-
-    return buttonsFrom(buttons);
+    return React.useMemo(() => {
```

Comment:
So React.useMemo shouldn't be used in a callback like this, i think this is what i was imagining:
```js
const buttonMarkup = React.useMemo(() => {
if (pages <= pageRange) {
     return _createButtons(0, pages);
    } else {
      return _createButtons(start, start + pageRange);
    }
}, [start, pages, pageRange])
```
Could even pull in `_createButtons` if you want

Reply:
Sorry for the confusion, added the initial comment to the wrong piece of code